core/utils/svg_paths.js

```diff
@@ -0,0 +1,153 @@
+  return command + points.join('');
+};
+/**
+ * Move the cursor to the given position without drawing a line.
+ * The coordinates are absolute.
+ * These coordinates are unitless and hence in the user coordinate system.
+ * See developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Paths#Line_commands
+ * @param {number} x The absolute x coordinate.
+ * @param {number} y The absolute y coordinate.
+ * @return {string} A string of the format ' M x,y '
+ * @public
+ */
+Blockly.utils.Paths.moveTo = function(x, y) {
+  return ' M ' + x + ',' + y + ' ';
+};
+/**
+ * Move the cursor to the given position without drawing a line.
+ * Coordinates are relative.
+ * These coordinates are unitless and hence in the user coordinate system.
+ * See developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Paths#Line_commands
+ * @param {number} dx The relative x coordinate.
+ * @param {number} dy The relative y coordinate.
+ * @return {string} A string of the format ' m dx,dy '
+ * @public
+ */
+Blockly.utils.Paths.moveBy = function(dx, dy) {
+  return ' m ' + dx + ',' + dy + ' ';
+};
+
+/**
+ * Draw a line from the current point to the end point, which is the current
+ * point shifted by dx along the x-axis and dy along the y-axis.
+ * These coordinates are unitless and hence in the user coordinate system.
+ * See developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Paths#Line_commands
+ * @param {number} dx The relative x coordinate.
+ * @param {number} dy The relative y coordinate.
+ * @return {string} A string of the format ' L dx,dy '
+ * @public
+ */
+Blockly.utils.Paths.lineTo = function(dx, dy) {
+  return ' l ' + dx + ',' + dy + ' ';
+};
+
+/**
+ * Draw multiple lines connecting all of the given points in order.  This is
+ * equivalent to a series of 'l' commands.
+ * These coordinates are unitless and hence in the user coordinate system.
+ * See developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Paths#Line_commands
+ * @param {!Array.<string>} points An array containing all of the points to
+ *     draw lines to, in order.  The points are represented as strings of the
+ *     format ' dx, dy '.
+ * @return {string} A string of the format ' l (dx,dy)+ '
+ * @public
+ */
+Blockly.utils.Paths.line = function(points) {
+  return 'l' + points.join('');
+};
+
+/**
+ * Draw a horizontal or vertical line.
+ * The first argument specifies the direction and whether the given position is
+ * relative or absolute.
+ * These coordinates are unitless and hence in the user coordinate system.
+ * See developer.mozilla.org/en-US/docs/Web/SVG/Attribute/d#LineTo_path_commands
+ * @param {string} command The command to prepend to the coordinate.  This
+ *     should be one of: V, v, H, h.  See
+ * @param {number} val The coordinate to pass to the command.  It may be
+ *     absolute or relative.
+ * @return {string} A string of the format ' command val '
+ * @public
+ */
+Blockly.utils.Paths.lineOnAxis = function(command, val) {
+  return command + ' ' + val;
```

Comment:
Again, did you want spaces on either side of the return value as specified in the JSDoc?

Reply:
Done